endpoints/openrtb2/amp_auction.go

```diff
@@ -578,21 +588,46 @@ func setAmpExt(site *openrtb2.Site, value string) {
 	}
 }
 
-func readPolicy(consent string) (privacy.PolicyWriter, error) {
-	if len(consent) == 0 {
+// setRegExtGDPR sets regs.ext.gdpr to either 0 or 1 only if it was set in the request query
+func setRegExtGDPR(gdprApplies *bool, req *openrtb2.BidRequest) error {
+	if gdprApplies == nil {
+		return nil
+	}
+
+	gdpr := int8(0)
+	if *gdprApplies {
+		gdpr++
+	}
+
+	reqWrap := &openrtb_ext.RequestWrapper{BidRequest: req}
+	if regsExt, err := reqWrap.GetRegExt(); err == nil {
+		regsExt.SetGDPR(&gdpr)
+	} else {
+		return err
+	}
+	return reqWrap.RebuildRequest()
+}
+
+func (pr policyReader) readPolicy(ampParams amp.Params, req *openrtb2.BidRequest) (privacy.PolicyWriter, error) {
```

Comment:
This is now amp specific, so please move it into the amp package.

Reply:
Good catch. Done